src/query/storage/promremote/storage.go

```diff
@@ -543,9 +543,11 @@ func (p *promStorage) write(
 			err = nil
 			break
 		}
-		p.retryWrites.Inc(1)
-		time.Sleep(backoff)
-		backoff *= 2
+		if i > 0 { // only do backoff if we are going to retry
+			p.retryWrites.Inc(1)
+			time.Sleep(backoff)
+			backoff *= 2
+		}
```

Comment:
How many Goroutines at most can you call `write()` concurrently? If there aren't enough Goroutines, retry backoffs may hold too many of them and starve other new requests.